Error Checker and Output Sheet Formatting

tvu212 opened the pull request:
Checks inputs to make sure they are valid, if they are not valid it will indicate what column and row number.

Resolves #48, resolves #39

tvu212 committed:
Update main.js

tvu212 committed:
Update main.js

tvu212 committed:
Update Matcher.js

tvu212 committed:
Update Student.js

tvu212 committed:
Update Workshop.js

beszel commented on main.js:
```diff
@@ -93,11 +115,31 @@ function main() {
+        }
+        if (typeof(grade) != "string"){
+            throw "Invalid input at row " + j + " column " + COLUMN_GRADE;
+        }
```
Can we move this into one or more functions, to be called here? Same for the name and capacity checks.

tvu212 replied:
Yeah sounds good Ill work on that today.

beszel commented on main.js:
```diff
@@ -42,8 +52,6 @@ var workshopData = WORKSHOP_SHEET.getDataRange().getValues();
 var outputSheet = RESPONSE_SPREADSHEET.getSheets()[OUTPUT_SHEET_INDEX];
 //var outputData = outputSheet.getDataRange().getValues(); Only needed for reading data
-//Clear the current output sheet
-outputSheet.clear();
```
Why aren't we clearing the output sheet anymore?

tvu212 replied:
I took this out because we already clear the output sheet in the main function

beszel commented on main.js:
```diff
@@ -73,14 +81,28 @@ function main() {
+        
+        //Checks name, number, and capacity, for valid inputs.//
+        
+        if (typeof(name) != "string"){
```
@tvu212 Can you use a lab computer to run the git hooks on this? The linter objects to `!=`, and prettify will clean up formatting.

We could also quickly set up your personal machine for development, if that would work better.

beszel commented on main.js:
```diff
@@ -73,14 +81,28 @@ function main() {
         var capacity = workshopData[i][COLUMN_WORKSHOP_CAPACITY];
+        var location = workshopData[i][COLUMN_WORKSHOP_BUILDING] + " " + workshopData[i][COLUMN_WORKSHOP_ROOM];
+        
+        //Checks name, number, and capacity, for valid inputs.//
```
This says it checks workshop number, but I only see name and capacity. Should there be something to check the number as well?

tvu212 replied:
the number will always be an int since it will be so I never saw the need to check i.

beszel reviewed (changes requested):
Can you add testing instructions in the PR? What steps do I need to follow to get these errors? Where should I look to see the results of these changes?

beszel committed:
Removed whitespace, fixed comments and formatting

tvu212 committed:
Update main.js

You can test by putting wrong inputs into the spreadsheet we are working on. While the checker does not check for all possible incorrect inputs, it does cover all the cases in which we need integers in some inputs to use in our class.

tvu212 committed:
Update main.js

Commit:
Its all good in the hood

beszel committed:
Move input checkers into their own file

Chris-Pyles reviewed (approved):
Looks great, ran compatibility tests with my stuff and it works seamlessly.

Might want to add a little more documentation to some functions but I think it's good to merge anyway, that can always be taken care of at any time

tvu212 committed:
Fixed integer input checks by including NaN conditions

beszel commented on error-checking.js:
```diff
@@ -0,0 +1,43 @@
+
+function preference_input_checker(workshopNum, j, k) {
+    Logger.log("Checking preferred workshop with type " + typeof workshopNum);
+    if ((typeof workshopNum !== "number") || (typeof workshopNum == NaN) {
```
Missing parenthesis

tvu212 replied:
Fixed them and checked for lynting errors

beszel reviewed (changes requested):
@tvu212 Again, it would be a big help if you can run linting before submitting for review. Also, `NaN` is a value, not a type. The type of `NaN` is `number`.

tvu212 committed:
Update error-checking.js

beszel reviewed (changes requested):
@tvu212 I'm still getting lint errors. Have you pushed all the latest changes?

Commit:
Fixed lynting errors

beszel committed:
Fix formatting

Commit:
Fixed the dumb indexing error

Commit:
Merge branch 'error_checker' of github.com:WhitmanCSCapstone/ge-scheduling into error_checker

beszel committed:
Remove log message

A frequently-logged message was cluttering the logs.

beszel committed:
Clarified error messages, rows, and columns

Also fixed comments and style, including function names

beszel commented:
@tvu212 I've pushed the latest stuff. If it looks okay, go ahead and merge this one.
